pages/account/addressBook.tsx

```diff
@@ -1,9 +1,182 @@
 import AccountBaseTemplate from "@/components/AccountBaseTemplate";
+import { Button } from "@/components/Button";
+import AddressDisplay from "@/components/checkout/AddressDisplay";
+import Spinner from "@/components/Spinner";
+import {
+  UserAddressesDocument,
+  AddressTypeEnum,
+  AddressDetailsFragment,
+  useSetAddressDefaultMutation,
+  useDeleteAddressMutation,
+  useUserAddressesQuery,
+} from "@/saleor/api";
+import { useAuthState } from "@saleor/sdk";
+import { Key } from "react";
 
 const AddressBookPage: React.VFC = () => {
+  const { authenticated } = useAuthState();
+  const { loading, error, data } = useUserAddressesQuery({
+    skip: !authenticated,
+    fetchPolicy: "network-only",
+  });
+  const [setAddressDefaultMutation] = useSetAddressDefaultMutation();
+  const [deleteAddressMutation] = useDeleteAddressMutation({
+    refetchQueries: [{ query: UserAddressesDocument }],
+  });
+  if (loading) {
+    return (
+      <AccountBaseTemplate>
+        <Spinner />
+      </AccountBaseTemplate>
+    );
+  }
+  if (error) return <p>Error : {error.message}</p>;
+
+  if (!data?.me?.addresses) {
+    return (
+      <AccountBaseTemplate>
+        No addresses information for this user
+      </AccountBaseTemplate>
+    );
+  }
+
+  let addresses = data?.me?.addresses || [];
+
+  const getAddressSection = (address: AddressDetailsFragment, index: Key) => {
```

Comment:
Instead of nested function, I would suggest creating new component: AddressBookCard with arguments 
- address: AddressFragment
- isDefaultShipping: boolean
- isDefaultBilling: boolean
- onRefreshBook: refresh callback

It would improve readability of this view.